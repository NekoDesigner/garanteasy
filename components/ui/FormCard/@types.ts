/**
 * IFormCardProps - Generated by Garanteasier CLI
 */

import { PropsWithChildren } from "react";

export interface IFormCardProps extends PropsWithChildren {
  style?: object;
  testID?: string;
}