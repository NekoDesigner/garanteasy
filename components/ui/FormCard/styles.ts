/**
 * FormCardStyles - Generated by Garanteasier CLI
 */

import { StyleSheet } from 'react-native';
import { COLORS } from '../../../constants';

const styles = StyleSheet.create({
  container: {
    gap: 10,
    backgroundColor: COLORS.greyLighter,
    borderColor: COLORS.greyDarker,
    borderWidth: 1,
    padding: 16,
    borderRadius: 8,
  }
});

export default styles;
