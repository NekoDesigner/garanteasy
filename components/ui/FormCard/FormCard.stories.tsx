/**
 * FormCard - Generated by Garanteasier CLI
 */

import React from "react";
import { View } from "react-native";
import FormCard from "./index";

export default {
  title: 'FormCard',
  component: FormCard
};

export const Default = () => (
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center', padding: 16, gap: 24 }}>
      <FormCard />
    </View>
);