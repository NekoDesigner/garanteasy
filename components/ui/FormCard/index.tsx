/**
 * FormCard - Generated by Garanteasier CLI
 */

import React from "react";
import { View } from "react-native";
import { IFormCardProps } from "./@types";
import FormCardStyles from "./styles";

const FormCard: React.FC<IFormCardProps> = ({style, testID = 'formcard', children, ...props}) => {
  return (
    <View style={[FormCardStyles.container, style]} testID={testID} {...props}>
      {children}
    </View>
  );
};

export default FormCard;
