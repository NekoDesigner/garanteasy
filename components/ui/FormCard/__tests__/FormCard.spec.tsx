/**
 * FormCard - Generated by Garanteasier CLI
 */

import { render } from "@testing-library/react-native";
import React from "react";
import FormCard from "../index";

describe("FormCard Components", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

    test('renders correctly', () => {
      const tree = render(<FormCard />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});