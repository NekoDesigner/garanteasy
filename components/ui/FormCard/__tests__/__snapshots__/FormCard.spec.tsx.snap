// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCard Components renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#F2F2F2",
        "borderColor": "#D9D9D9",
        "borderRadius": 8,
        "borderWidth": 1,
        "gap": 10,
        "padding": 16,
      },
      undefined,
    ]
  }
  testID="formcard"
/>
`;
